src/cell_ref.rs

```diff
@@ -26,16 +29,19 @@ where
     ///
     /// This method allows handling of reference overflows, but:
     ///
-    /// * Having 4 billion / 9 quintillion references to an object is not a
+    /// * Having 2 billion / 9 quintillion references to an object is not a
     ///   realistic scenario in most applications.
     /// * Applications that hold `CellRef`s with an ever-increasing reference
-    ///   count is not supported by this library.
+    ///   count are not supported by this library.
     ///
-    ///     Reaching `usize::MAX` may be possible with
+    ///     Reaching `isize::MAX` may be possible with
     ///     `std::mem::forget(CellRef::clone(&r))`.
     pub fn try_clone(&self) -> Result<Self, RefOverflow> {
-        let previous_value = self.flag.fetch_add(1, Ordering::Release);
-        if previous_value == usize::MAX {
+        let previous_value = self.flag.fetch_add(1, Ordering::Relaxed);
+
+        let overflow = previous_value == REF_LIMIT_MAX;
+        if unlikely(overflow) {
+            self.flag.fetch_sub(1, Ordering::SeqCst);
```

Comment:
It's not important as branch is unlikely. But `Relaxed` semantics should be enough. This decrement does not synchronize anything.